stock_picking_consumable_effective_date/models/stock_picking.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    show_consumable_date = fields.Boolean(compute="_compute_show_consumable_date")
+
+    def _compute_show_consumable_date(self):
+        for pick in self:
+            pick.show_consumable_date = False
+            if pick.picking_type_code not in ("incoming", "outgoing"):
+                continue
+            if pick.move_ids.with_company(pick.company_id).product_id.filtered(
+                lambda x: x.detailed_type == "consu"
+            ):
```

Comment:
We do not want to show both "Accounting Date" and "Consumable Effective Date" when a picking contains stockable and consumable products.

```suggestion
            moves = pick.move_ids
            # When all moves are consumables.
            if moves and not moves.with_company(pick.company_id).product_id.filtered(
                lambda x: x.detailed_type == "product"
            ):
```